src/CESMII.OpcUa.CloudLibraryResolver/CloudLibraryResolver.cs

```diff
@@ -111,7 +111,10 @@ public async Task<IEnumerable<String>> ResolveNodeSetsAsync(List<ModelNameAndVer
                 var bestMatch = nodesetWithURIAndDate.FirstOrDefault(n => n?.Item1 == missing.ModelUri && n?.Item2 == missing.PublicationDate);
                 if (bestMatch == null)
                 {
-                    bestMatch = nodesetWithURIAndDate.Where(n => n?.Item1 == missing.ModelUri && (missing.PublicationDate == null || n?.Item2 >= missing.PublicationDate)).OrderBy(m => m?.Item2).FirstOrDefault();
+                    if (missing.PublicationDate==null)
+                        bestMatch = nodesetWithURIAndDate.Where(n => n?.Item1 == missing.ModelUri && (missing.PublicationDate == null || n?.Item2 >= missing.PublicationDate)).OrderByDescending(m => m?.Item2).FirstOrDefault();
```

Comment:
Can drop the missing.PublicationDat == null check in line 115. It's always true.